apps/condo/domains/resident/utils/serverSchema/findOrganizationsForAddress.js

```diff
@@ -0,0 +1,122 @@
+                    period,
+                    account: { id_in: Object.keys(billingAccountsNumbersIndex) },
+                })
+                if (billingReceipts.length) {
+                    receipts = billingReceipts.map(receipt => ({
+                        organizationId: id,
+                        number: billingAccountsNumbersIndex[receipt.account],
+                        category: receipt.category,
+                        balance: Big(receipt.toPay).toFixed(2),
+                        routingNumber: get(receipt, 'recipient.bic'),
+                        bankAccountNumber: get(receipt, 'recipient.bankAccount'),
+                    }))
+                }
+            }
+        }
+        const accountNumber = accountQuery.number
+        if (!receipts.length && checkAccountNumberUrl && accountNumber) {
+            const { status, services } = await getAccountsWithOnlineInteractionUrl(checkAccountNumberUrl, tin, accountNumber)
+            if (status === ONLINE_INTERACTION_CHECK_ACCOUNT_SUCCESS_STATUS) {
+                receipts = services.map(service => {
+                    const { category, account: { number }, bankAccount: { number: bankAccountNumber, routingNumber } } = service
+                    return {
+                        organizationId: id,
+                        category,
+                        number,
+                        routingNumber,
+                        bankAccountNumber,
+                    }
+                })
+            }
+        }
+        return receipts
+    }))
+    return receipts.flat()
+}
+
+async function getBillingInformationForOrganizations (organizations) {
+    const billingContexts = await find('BillingIntegrationOrganizationContext', {
+        status: BILLING_CONTEXT_FINISHED_STATUS,
+        deletedAt: null,
+        organization: { id_in: organizations.map(({ id }) => id) },
+    })
+    const billingContextIndex = Object.fromEntries(billingContexts.map(context => ([context.organization, context])))
+    const billingIntegrationsWithRemoteInteraction = await find('BillingIntegration', {
+        id_in: billingContexts.map(({ integration }) => integration),
+        checkAccountNumberUrl_not: null,
+    })
+    const remoteInteractions = Object.fromEntries(billingIntegrationsWithRemoteInteraction
+        .map(({ id, checkAccountNumberUrl }) => ([ id, checkAccountNumberUrl])))
+
+    return Object.fromEntries(organizations.map(organization => {
+        const context = billingContextIndex[organization.id]
+        if (!context) {
+            return [organization.id, null]
+        }
+        return [organization.id, {
+            id: context.id,
+            period: get(context, 'lastReport.period'),
+            checkAccountNumberUrl: remoteInteractions[context.integration],
+        }]
+    }))
+}
+
+async function getOrganizationsWithMeters (organizationIds, query = {}) {
+    const organizationsWithMeters = await Promise.all(organizationIds.map(async id => {
```

Comment:
Also bad logic, MeterResourceOwner rules bypassed here!

Reply:
We only find organizations for address. We check only that organization has any meters